lib/Sema/TypeCheckMacros.cpp

```diff
@@ -357,6 +360,24 @@ bool ExpandSynthesizedMemberMacroRequest::evaluate(Evaluator &evaluator,
   return synthesizedMembers;
 }
 
+bool ExpandPeerMacroRequest::evaluate(Evaluator &evaluator, Decl *decl) const {
+  SmallVector<Decl *, 2> peers;
+  decl->forEachAttachedMacro(MacroRole::Peer,
+      [&](CustomAttr *attr, MacroDecl *macro) {
+        expandPeers(attr, macro, decl, peers);
+      });
+
+  // Expand all peers that have attached peer macros.
+  for (auto *peer : peers) {
```

Comment:
This makes me nervous for reasons I can't quite enumerate. It feels a bit ad hoc, I guess, putting the recursion into the request itself vs. happening because some outer walk sees the new peers and expands their peers.

Reply:
I've removed this for now. I think we should ban it, because nested peer expansion feels really problematic under a lazy model that expands based on names that are "covered" by macro attributes.